### PR TITLE
Create application now returns the key value in a struct with the ID

### PR DESCRIPTION
**Pull request by PCloughster**
Create application wasn't previously returning the key for the API key created, needed for terraform provider

### CODE REVIEW

**Comment by PCloughster** [comment]
Added UpdateApplication function to allow updating of application name and description

File: pkg/service/account/service_application.go [hunk 1]
```diff
@@ -56,14 +56,29 @@ func (s *Service) getServicesPaginatedResponseBody(parameters connection.APIRequ
...
+
+func (s *Service) UpdateApplication(req CreateApplicationRequest) (string, error) {
+	body, err := s.updateApplicationResponseBody(req)
+
 	return body.Data.ID, err
 }
 
-func (s *Service) createApplicationResponseBody(req CreateApplicationRequest) (*connection.APIResponseBodyData[Application], error) {
-	return connection.Post[Application](s.connection, "/account/v1/applications", &req)
+func (s *Service) updateApplicationResponseBody(req CreateApplicationRequest) (*connection.APIResponseBodyData[Application], error) {
```
Comment (0x4c6565) [inline pkg/service/account/service_application.go:80]: You're accepting `req CreateApplicationRequest` for update here

File: pkg/service/account/service_application.go [hunk 2]
```diff
@@ -56,14 +56,29 @@ func (s *Service) getServicesPaginatedResponseBody(parameters connection.APIRequ
 	return body, response.HandleResponse(body, nil)
 }
 
-func (s *Service) CreateApplication(req CreateApplicationRequest) (string, error) {
+func (s *Service) CreateApplication(req CreateApplicationRequest) (CreateApplicationResponse, error) {
 	body, err := s.createApplicationResponseBody(req)
 
+	response := CreateApplicationResponse{
```
Comment (0x4c6565) [inline pkg/service/account/service_application.go:62]: Rather than create a new response here, you can just return `body.Data`

**Review by 0x4c6565 — CHANGES_REQUESTED** [review]

**Review by 0x4c6565 — APPROVED** [review]
LGTM

**Pull request merged**